### PR TITLE
added `State.DELETED`

### PR DESCRIPTION
**Pull request by thisismana**
- also some documentation improvements (added introduction from README.md)

### CODE REVIEW

**Review by bobaaaaa — APPROVED** [review]

File: stroeer/core/v1/article.proto [hunk 1]
```diff
@@ -411,23 +411,28 @@ message Metadata {
...
   * }
   * ```
   */
   enum State {
-    // Not specified.
     STATE_UNSPECIFIED = 0;
-
-    // A published article.
     PUBLISHED = 1;
+    DELETED = 2;
```
Comment (moritzzimmer) [inline stroeer/core/v1/article.proto:435]: The cms api responds with

```json
{
"state": "archived"
}
```

for articles which have been deleted (label of the CMS button). Guess combining this into `State.DELETED` is fine.

File: stroeer/core/v1/article.proto [hunk 2]
```diff
@@ -411,23 +411,28 @@ message Metadata {
...
-  * State of the article in the content management system.
+  *
+  * State of the item ([`Article`](article.html), [`Element`](element.html))
+  * in the content management system. The `state` in combination with
+  * [start_time](#start_time) and [end_time](#end_time) determines whether or not
+  * this item should be rendered; this must be respected by all consumers especially
+  * when content is duplicated or cached.
+  *
+  * The terms `deleted` (articles) and `archived` (media lib) are interchangeable/synonyms.
+  * This API combines those two into `State.DELETED`.
```
Comment (moritzzimmer) [inline stroeer/core/v1/article.proto:422]:
```suggestion
  * This message combines those two into `State.DELETED`.
```

File: stroeer/core/v1/article.proto [hunk 3]
```diff
@@ -411,23 +411,28 @@ message Metadata {
...
+  * enum State {
   *   STATE_UNSPECIFIED = 0;
-  *
-  *   // A published article.
   *   PUBLISHED = 1;
+  *   DELETED = 2;
   * }
   * ```
   */
   enum State {
```
Comment (moritzzimmer) [inline stroeer/core/v1/article.proto:432]: It's possible to unpublish articles in the cms as well - shall we add `UNPUBLISHED`, too?

File: stroeer/core/v1/article.proto [hunk 4]
```diff
@@ -411,23 +411,28 @@ message Metadata {
...
-    // Not specified.
```
Comment (moritzzimmer) [inline stroeer/core/v1/article.proto:426]: We could add comments describing the expected grpc/http status codes for those states

**Review by moritzzimmer — CHANGES_REQUESTED** [review]

File: stroeer/core/v1/article.proto (hunk 3 above)
Comment (thisismana) [reply stroeer/core/v1/article.proto:436]: added `DRAFT` for this purpose

File: stroeer/core/v1/article.proto (hunk 4 above)
Comment (thisismana) [reply stroeer/core/v1/article.proto:426]: those can be documented in the service.proto, I guess. This has not been done until now.

I guess copying the essence of https://github.com/stroeer/paper/issues/757 over would be nice

**Review by moritzzimmer — APPROVED** [review]

**Pull request merged**